### PR TITLE
Genetics Nerf; Butchers Space Adaption Into a Two Gene Combo (Plus a New Gene!)

### PR DESCRIPTION
**Pull request by Jolly-66**
I really fucking hate genetics as it stands, and all of science for that matter, so this is my band-aid """fix""" to nerfing it.

Space adaption is NO LONGER a printable mutation, instead its a combo of two new genes.

FRIGID IMMUNITY
Cold resistance. 10 instability.
You will take x2 more heat related damage.

SWEALTERING IMMUNITY
Heat resistance. 10 instability.
You will now take x2 more cold related damage.

VACUUM ADAPTATION
Vacuum resistance. 30 instability.
You will now take x2.5 more oxy loss related damage (non suffocation related).
You will now have a x2.25 pain modifier. 

Vacuum Adaptation + Frigid Immunity = Space adaptation. 

A SMALL THING TO NOTE:
For some reason the "halo" isn't appearing when you combo these genes, will look into it sometime.

### CODE REVIEW

File: code/datums/mutations/space_adaptation.dm [hunk 1]
```diff
@@ -7,6 +7,7 @@
 	text_gain_indication = "<span class='notice'>Your body feels warm!</span>"
 	time_coeff = 5
 	instability = 30
+	locked = TRUE //NON-MODULAR CHANGE; should block this mutation from appearing in the consoles
```
Comment (MrMelbert) [inline code/datums/mutations/space_adaptation.dm:10]: This can be done modularly

**Comment by CRITAWAKETS** [comment]
Why are we even keeping space adaptation if cold resistance + vacuum resistance does the same thing?

Also, cold resistance and vacuum resistance should have their instability values swapped. You encounter cold a lot more than vacuum and its generally more deadly (especially to them lizerds and slimeys). I'd recommend changing it though so that cold resistance is 20 and vacuum is 10, unless you want to up the instability cost.

Also this really isnt a nerf especially since heat resistance is a thing. If anything, its a buff.

**Comment by Jolly-66** [comment]
> Also, cold resistance and vacuum resistance should have their instability values swapped. You encounter cold a lot more than vacuum and its generally more deadly (especially to them lizerds and slimeys). I'd recommend changing it though so that cold resistance is 20 and vacuum is 10, unless you want to up the instability cost.

Should we just bump both temps to 15~20 with vacuum 25~30?

@MrMelbert

File: jollystation_modules/code/datums/mutations/cold_adaptation.dm [hunk 2]
```diff
@@ -0,0 +1,22 @@
+//Cold immunity
+/datum/mutation/human/cold_adaptation
+	name = "Frigid Adaptation"
+	desc = "A pecuilar mutation that allows the host to be completely resistant to absolute zero."
+	quality = POSITIVE
+	difficulty = 32
+	text_gain_indication = "<span class='notice'>Your body feels warm!</span>"
+	time_coeff = 5
+	instability = 10
```
Comment (MrMelbert) [inline jollystation_modules/code/datums/mutations/cold_adaptation.dm:9]:
```suggestion
	instability = 20
```

File: jollystation_modules/code/datums/mutations/hot_adaptation.dm [hunk 3]
```diff
@@ -0,0 +1,22 @@
+//Heat immunity
+/datum/mutation/human/heat_adaptation
+	name = "Swealtering Adaptation"
+	desc = "A pecuilar mutation that allows the host to be completely resistant to the hottest tempeatures known to man."
+	quality = POSITIVE
+	difficulty = 32
+	text_gain_indication = "<span class='notice'>Your body feels cool!</span>"
+	time_coeff = 5
+	instability = 10
```
Comment (MrMelbert) [inline jollystation_modules/code/datums/mutations/hot_adaptation.dm:9]:
```suggestion
	instability = 20
```

**Review by MrMelbert — COMMENTED** [review]
In my opinion:

File: jollystation_modules/code/datums/mutations/hot_adaptation.dm [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+	if(..())
+		return
+	ADD_TRAIT(owner, TRAIT_RESISTHEAT, GENETIC_MUTATION)
+	owner.physiology.heat_mod *= 2
+
+/datum/mutation/human/heat_adaptation/on_losing(mob/living/carbon/human/owner)
+	if(..())
+		return
+	REMOVE_TRAIT(owner, TRAIT_RESISTHEAT, GENETIC_MUTATION)
+	owner.physiology.heat_mod /= 2
```
Comment (MrMelbert) [inline jollystation_modules/code/datums/mutations/hot_adaptation.dm:22]: I think you got these backwards. Heat adaptation should give you a positive cold_mod, and cold visa-versa. 

You're making them immune to heat, then making them more vulnerable to heat, which doesn't do anything

**Comment by MrMelbert** [comment]
I'm fine with it being split, every mutation balances genetics slightly more

For number references:

Lizards have a 1.5x cold modifier and 0.67 heat modifier.
Plasmamen and podpeople have a 1.5x heat modifier.

File: jollystation_modules/code/datums/mutations/vacuum_immunity.dm [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+	time_coeff = 5
+	instability = 30
+
+
+/datum/mutation/human/vacuum_adaptation/on_acquiring(mob/living/carbon/human/owner)
+	if(..())
+		return
+	ADD_TRAIT(owner, TRAIT_RESISTLOWPRESSURE, GENETIC_MUTATION)
+	owner.physiology.oxy_mod *= 2.5
+	owner.set_pain_mod(PAIN_MOD_QUIRK, 1.75)
```
Comment (MrMelbert) [inline jollystation_modules/code/datums/mutations/vacuum_immunity.dm:17]: Don't use the quirk pain modifier for this
Define a new modifier (pain_modifier_genetic or something)
and in `proc/remove_all_pain`, you should make sure that it doesn't get removed

File: jollystation_modules/code/datums/mutations/cold_adaptation.dm [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+	if(..())
+		return
+	ADD_TRAIT(owner, TRAIT_RESISTCOLD, GENETIC_MUTATION)
+	owner.physiology.cold_mod *= 2
+
+/datum/mutation/human/cold_adaptation/on_losing(mob/living/carbon/human/owner)
+	if(..())
+		return
+	REMOVE_TRAIT(owner, TRAIT_RESISTCOLD, GENETIC_MUTATION)
+	owner.physiology.heat_mod /= 2
```
Comment (MrMelbert) [inline jollystation_modules/code/datums/mutations/cold_adaptation.dm:22]: cold mod *=2 
heat mod /=2

🤔

Comment (Jolly-66) [reply jollystation_modules/code/datums/mutations/cold_adaptation.dm]: when melbert reviews the same file twice

sus

File: jollystation_modules/code/datums/mutations/vacuum_immunity.dm (hunk 5 above)
Comment (Jolly-66) [reply jollystation_modules/code/datums/mutations/vacuum_immunity.dm:17]: I don't know how you want me to make sure it doesn't get removed; I'm looking at the code and I'm exceptionally mush brained

File: jollystation_modules/code/datums/mutations/cold_adaptation.dm (hunk 6 above)
Comment (MrMelbert) [reply jollystation_modules/code/datums/mutations/cold_adaptation.dm]: i didn't review the same file twice
i reviewed heat adaptation to say "you got it backwards" and cold adaptation to say "you copypasted wrong"

**Review by MrMelbert — APPROVED** [review]
mutations / modifiers could use some numbers tweaks still but I think it's fine now

**Comment by Jolly-66** [comment]
I'm gonna figure out the halo bullshit later, probably gonna do a hack fix but its whatever.

**Pull request merged**